lesson17.py

```diff
@@ -0,0 +1,139 @@
+# Task 1
+
+class Animal:
+
+    def talk(self):
+        pass
+
+
+class Dog(Animal):
+    def talk(self):
+        return "Woof, Woof"
+
+
+class Cat(Animal):
+    def talk(self):
+        return "Meow"
+
+
+def generic_func(instance_class):
+    if isinstance(instance_class, Animal):
+        return instance_class.talk()
+
+
+dog = Dog()
+cat = Cat()
+
+print("Dog say:", generic_func(dog))
+print("Cat say:", generic_func(cat))
+
+
+# Task 2
+
+
+class Author:
+    def __init__(self, name, country, birthday):
+        self.name = name
+        self.country = country
+        self.birthday = birthday
+        self.books = []

```

Comment:
This attribute is not processed anywhere in code. I suggest to append it in `Book`'s `__init__`